src/win/process.c

```diff
@@ -249,19 +249,16 @@ static WCHAR* path_search_walk_ext(const WCHAR *dir,
                                    const WCHAR *name,
                                    size_t name_len,
                                    WCHAR *cwd,
-                                   size_t cwd_len,
-                                   int name_has_ext) {
+                                   size_t cwd_len) {
   WCHAR* result;
 
-  /* If the name itself has a nonempty extension, try this extension first */
-  if (name_has_ext) {
-    result = search_path_join_test(dir, dir_len,
-                                   name, name_len,
-                                   L"", 0,
-                                   cwd, cwd_len);
-    if (result != NULL) {
-      return result;
-    }
+  /* Try the name itself first */
+  result = search_path_join_test(dir, dir_len,
+                                 name, name_len,
+                                 L"", 0,
+                                 cwd, cwd_len);
```

Comment:
This searches the `PATH` for any file with the exact name.  When CYGWIN or MSYS tools are in the `PATH` on Windows then we may find wrapper shell scripts, without any extension, and be unable to run them since our calling process is not built against the cygwin or msys2 runtime.